src/components/molecules/NavBar/NavBar.tsx

```diff
@@ -89,14 +89,18 @@ const NavBar: React.FC<NavBarPropsType> = ({ redirectionUrl }) => {
   const radioStations = useSelector(radioStationsSelector);
   const parentVenue = useSelector(parentVenueSelector);
 
-  const venueParentId = venue?.parentId;
-  const venueParentQuery = useMemo<ReduxFirestoreQuerySetting>(
-    () => ({
-      collection: "venues",
-      doc: venueParentId,
-      storeAs: "parentVenue",
-    }),
-    [venueParentId]
+  const venueParentQuery = useMemo<SparkleDocRFQSetting[]>(
+    () =>
+      venue?.parentId
+        ? [
+            {
+              collection: "venues",
+              doc: venue.parentId,
+              storeAs: "parentVenue",
+            },
+          ]
+        : [],
+    [venue]
```

Comment:
We don't actually need to `useMemo` the queries as it does a deep compare.

It could also be cool to have a different param on our `useFirestoreConnect` overloaded function for handling these 'check boolean and only then give the config', as it's a pattern we use a lot.

Something like (names are hard):

```
useSparkleFirestoreConnectIfDefined(booleanCheck, maybeConfig)
```